// ng2-smart-table/esm5/lib/components/filter/filter-types/default-filter.js
import { __decorate, __metadata } from "tslib";
import { Input, Output, EventEmitter, Component } from '@angular/core';
import { Column } from '../../../lib/data-set/column';
var DefaultFilter = /** @class */ (function () {
    function DefaultFilter() {
        this.delay = 300;
        this.filter = new EventEmitter();
    }
    DefaultFilter.prototype.ngOnDestroy = function () {
        if (this.changesSubscription) {
            this.changesSubscription.unsubscribe();
        }
    };
    DefaultFilter.prototype.setFilter = function () {
        this.filter.emit(this.query);
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], DefaultFilter.prototype, "query", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], DefaultFilter.prototype, "inputClass", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Column)
    ], DefaultFilter.prototype, "column", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], DefaultFilter.prototype, "filter", void 0);
    DefaultFilter = __decorate([
        Component({
            template: ''
        })
    ], DefaultFilter);
    return DefaultFilter;
}());
export { DefaultFilter };
